api/src/main/java/org/apache/iceberg/transforms/Truncate.java

```diff
@@ -71,16 +120,55 @@ public boolean preservesOrder() {
     return true;
   }
 
-  private static class TruncateInteger extends Truncate<Integer> {
-    private final int width;
+  @Override
+  public boolean satisfiesOrderOf(Transform<?, ?> other) {
+    if (this == other) {
+      return true;
+    }
+
+    if (!(other instanceof Truncate)) {
+      return false;
+    }
+
+    Truncate<?> otherTrunc = (Truncate<?>) other;
+    return otherTrunc.width <= width;
+  }
+
+  @Override
+  public boolean equals(Object o) {
+    if (this == o) {
+      return true;
+    } else if (!(o instanceof Truncate)) {
+      return false;
+    }
+
+    Truncate<?> that = (Truncate<?>) o;
+    return width == that.width;
+  }
+
+  @Override
+  public int hashCode() {
+    return Objects.hashCode(width);
+  }
 
+  @Override
+  public String toString() {
+    return "truncate[" + width + "]";
+  }
+
+  private static class TruncateInteger extends Truncate<Integer>
+      implements Function<Integer, Integer> {
     private TruncateInteger(int width) {
-      this.width = width;
+      super(width);
     }
 
     @Override
-    public Integer width() {
-      return width;
+    public Function<Integer, Integer> bind(Type type) {
+      if (type.typeId() == Type.TypeID.INTEGER) {
+        return this;
+      } else {
+        return null;
```

Comment:
What's the contract for `bind` if the type is not supported? I see some places return null and other places throw an exception.

Reply:
I think it should be to throw an exception. I should update this.

Reply:
Fixed.